packages/devtools_app/lib/src/standalone_ui/vs_code/debug_sessions.dart

```diff
@@ -86,53 +73,119 @@ class _DevToolsMenu extends StatelessWidget {
-            // TODO(dantup): Make these conditions use the real screen
-            //  conditions and/or verify if these conditions are correct.
-            _devToolsButton(
-              ScreenMetaData.inspector,
-              enabled: isFlutter && isDebug,
-            ),
-            _devToolsButton(
-              ScreenMetaData.cpuProfiler,
-              enabled: isDebug || isProfile,
-            ),
-            _devToolsButton(
-              ScreenMetaData.memory,
-              enabled: isDebug || isProfile,
-            ),
-            _devToolsButton(
-              ScreenMetaData.performance,
-            ),
-            _devToolsButton(
-              ScreenMetaData.network,
-              enabled: isDebug,
-            ),
-            _devToolsButton(
-              ScreenMetaData.logging,
-            ),
-            // TODO(dantup): Check other screens (like appSize) work embedded and
-            //  add here.
-          ],
-          child: const Text('DevTools'),
-        ),
+        if (api.capabilities.openDevToolsPage)
+          _DevToolsMenu(
+            api: api,
+            session: session,
+            isFlutter: isFlutter,
+            isDebug: isDebug,
+            isProfile: isProfile,
+          ),
       ],
     );
   }
+}
 
-  Widget _devToolsButton(
-    ScreenMetaData screen, {
-    bool enabled = true,
-  }) {
-    return IconButton(
-      onPressed: enabled
-          ? () => unawaited(api.openDevToolsPage(session.id, screen.id))
-          : null,
-      tooltip: screen.title ?? screen.id,
-      icon: Icon(screen.icon, size: actionsIconSize),
+class _DevToolsMenu extends StatelessWidget {
+  const _DevToolsMenu({
+    required this.api,
+    required this.session,
+    required this.isFlutter,
+    required this.isDebug,
+    required this.isProfile,
+  });
+
+  final VsCodeApi api;
+  final VsCodeDebugSession session;
+  final bool isFlutter;
+  final bool isDebug;
+  final bool isProfile;
+
+  @override
+  Widget build(BuildContext context) {
+    final normalDirection = Directionality.of(context);
+    final reversedDirection = normalDirection == TextDirection.ltr
+        ? TextDirection.rtl
+        : TextDirection.ltr;
+
+    Widget devToolsButton(
+      ScreenMetaData screen, {
+      bool enabled = true,
+    }) {
+      return SizedBox(
+        width: double.infinity,
+        child: TextButton.icon(
+          style: TextButton.styleFrom(
+            alignment: Alignment.centerRight,
+            shape: const ContinuousRectangleBorder(),
+          ),
+          onPressed: enabled
+              ? () => unawaited(api.openDevToolsPage(session.id, screen.id))
+              : null,
+          label: Directionality(
+            textDirection: normalDirection,
+            child: Text(screen.title ?? screen.id),
+          ),
+          icon: Icon(screen.icon, size: actionsIconSize),
+        ),
+      );
+    }
+
+    return Directionality(
+      textDirection: reversedDirection,
```

Comment:
why do we need this directionality widget?

Reply:
It's to have the menu anchor on the right and expand left (and also to put the icons on the right). I've added a comment to make this clearer.

![Screenshot 2023-09-15 105539](https://github.com/flutter/devtools/assets/1078012/bb420aa0-8a88-4b87-8a9a-9f96b88305e4)

![Screenshot 2023-09-15 105549](https://github.com/flutter/devtools/assets/1078012/b9239d51-d754-43f2-aacc-6c5626d4aedf)